### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by MorganLove**
**DESCRIPTION**

- Upgrades to react 16.9.0, react-dom 16.9.0, blueprintjs 3.20.0, & babel-jest 24.9.0
- Updating tests that rely on mocking `Date.now()` to work despite something in react-dom 16.9.0 messing with that.
- Cleaning up `act()` warnings now that we can use the async version with react 16.9
- Getting rid of the canvas warning from tests

**New Tests**

- More complete testing on the `poll` function in utilities.

**Checklist**

*Function*

[*] Code is fully functional and ready for review (or is tagged WIP)

[*] No errors/warnings in the browser console ( ﾟヮﾟ)/

[*] package.json is committed if I added or removed any dependencies

[*] Tests added as needed to ensure code base still has 100% passing coverage

[*] Tests added as needed to cover new functionality

*Style*

[*] Passes linter without errors

[*] Code is easily understandable and self-documenting (using naming conventions in keeping with the rest of the codebase)

[*] Comments are only included as needed to explain the 'why' instead of the 'what'

[*] Common standards and coding practices are adhered to

[*] DRY: Don't Repeat Yourself. Code is reasonably structured to avoid repetition

[*] Code is intuitively structured following a Separation of Concerns

[*] No class components; only functional components (using Hooks as needed)

### CODE REVIEW

File: arlo-client/src/components/utilities.test.ts [hunk 1]
```diff
@@ -56,4 +57,59 @@ describe('utilities.ts', () => {
...
+      const error = async () => (result = 'an error')
+      await poll(condition, callback, error)
+      await wait(() => {
+        expect(result).toBe('over five')
+      })
+    })
+
+    it('times out', async () => {
+      const j = (function* idMaker() {
+        var index = 1
```
Comment (eventualbuddha) [inline arlo-client/src/components/utilities.test.ts]:
```suggestion
        let index = 1
```

File: arlo-client/src/components/utilities.test.ts [hunk 2]
```diff
@@ -56,4 +57,59 @@ describe('utilities.ts', () => {
       expect(testNumber(50)(100)).resolves.toBe('Must be smaller than 50')
     })
   })
+
+  describe('poll', () => {
+    it('iterates', async () => {
+      const j = (function* idMaker() {
+        var index = 0
```
Comment (eventualbuddha) [inline arlo-client/src/components/utilities.test.ts]:
```suggestion
        let index = 0
```

File: arlo-client/src/components/utilities.test.ts [hunk 3]
```diff
@@ -56,4 +57,59 @@ describe('utilities.ts', () => {
...
+      const dateSpy = jest
+        .spyOn(Date, 'now')
+        .mockReturnValueOnce(startDate)
+        .mockReturnValueOnce(lateDate)
+      const j = (function* idMaker() {
+        var index = 1
+        while (true) yield index++
+      })()
+      let result = ''
+      const condition = async () => j.next().value < 1
```
Comment (eventualbuddha) [inline arlo-client/src/components/utilities.test.ts]:
```suggestion
      const condition = async () => false
```

We never want this condition to resolve to `true`, so I don't think we need the whole `idMaker` thing.

File: arlo-client/src/components/utilities.test.ts [hunk 4]
```diff
@@ -56,4 +57,59 @@ describe('utilities.ts', () => {
...
+
+  describe('poll', () => {
+    it('iterates', async () => {
+      const j = (function* idMaker() {
+        var index = 0
+        while (true) yield index++
+      })()
+      let result = ''
+      const condition = async () => j.next().value > 2
+      const callback = async () => (result = 'over five')
```
[the remark this answers is not in the document]
Comment (MorganLove) [reply arlo-client/src/components/utilities.test.ts]: It used to be testing over five, but I changed it partway through writing the tests and didn't update that, haha.

File: arlo-client/src/components/AuditForms/CalculateRiskMeasurement.test.tsx [hunk 5]
```diff
@@ -200,37 +210,20 @@ describe('CalculateRiskMeasurement', () => {
...
-      await wait(() => {
-        expect(dateSpy).toBeCalledTimes(2)
-        expect(apiMock).toBeCalled()
-        expect(setIsLoadingMock).toBeCalledTimes(1)
-        expect(getStatusMock).toBeCalled()
-        expect(updateAuditMock).toBeCalledTimes(0)
-        expect(toastSpy).toBeCalledTimes(1)
-      })
-    }
+    global.Date.now = realDate
```
[the remark this answers is not in the document]
Comment (MorganLove) [reply arlo-client/src/components/AuditForms/CalculateRiskMeasurement.test.tsx]: The original thrust behind this PR was to figure out why tests relating to the `Date.now()` function were suddenly failing when `react-dom` was upgraded to `9.0`. It was behaving very strangely and the spy was being inconsistent, so I tried this as part of the troubleshooting. I eventually found a workaround, but never went and put this back as a spy.

**Comment by MorganLove** [comment]
@eventualbuddha Thanks for the review! I pushed another commit that incorporates your feedback.

**Pull request merged**